godot-ffi/src/binding/mod.rs

```diff
@@ -0,0 +1,328 @@
+    let binding = unsafe { get_binding() };
+
+    debug_assert!(
+        !binding.class_editor_method_table.is_initialized(),
+        "server method table should only be initialized once"
+    );
+
+    // SAFETY: Is only called once, and is called before any accesses to this table.
+    unsafe { binding.class_server_method_table.set(table) }
+}
+
+/// # Safety
+///
+/// - The Godot binding must have been initialized before calling this function.
+/// - Must only be called once.
+///
+/// If the "experimental-threads" is not enabled then this must be called from the same thread that the bindings were initialized from.
+pub(crate) unsafe fn initialize_class_scene_method_table(table: ClassSceneMethodTable) {
+    // SAFETY: `get_binding` has the same preconditions as this function.
+    let binding = unsafe { get_binding() };
+
+    debug_assert!(
+        !binding.class_scene_method_table.is_initialized(),
+        "scene method table should only be initialized once"
+    );
+
+    // SAFETY: Is only called once, and is called before any accesses to this table.
+    unsafe { binding.class_scene_method_table.set(table) }
+}
+
+/// # Safety
+///
+/// - The Godot binding must have been initialized before calling this function.
+/// - Must only be called once.
+///
+/// If the "experimental-threads" is not enabled then this must be called from the same thread that the bindings were initialized from.
+pub(crate) unsafe fn initialize_class_editor_method_table(table: ClassEditorMethodTable) {
+    // SAFETY: `get_binding` has the same preconditions as this function.
+    let binding = unsafe { get_binding() };
+
+    debug_assert!(
+        !binding.class_editor_method_table.is_initialized(),
+        "editor method table should only be initialized once"
+    );
+
+    // SAFETY: Is only called once, and is called before any accesses to this table.
+    unsafe { binding.class_editor_method_table.set(table) }
+}
+
+/// # Safety
+///
+/// - The Godot binding must have been initialized before calling this function.
+/// - Must only be called once.
+///
+/// If the "experimental-threads" is not enabled then this must be called from the same thread that the bindings were initialized from.
+pub(crate) unsafe fn initialize_builtin_method_table(table: BuiltinMethodTable) {
+    // SAFETY: `get_binding` has the same preconditions as this function.
+    let binding = unsafe { get_binding() };
+
+    debug_assert!(
+        !binding.builtin_method_table.is_initialized(),
+        "builtin method table should only be initialized once"
+    );
+
+    // SAFETY: Is only called once, and is called before any accesses to this table.
+    unsafe { binding.builtin_method_table.set(table) }
+}
+
+/// # Safety
+///
+/// The Godot binding must have been initialized before calling this function.
+///
+/// If the "experimental-threads" is not enabled then this must be called from the same thread that the bindings were initialized from.
```

Comment:
```suggestion
/// If "experimental-threads" is not enabled, then this must be called from the same thread that the bindings were initialized from.
```

Maybe search&replace some others, too 🙂